$MirOS: ports/print/cups/patches/patch-backend_parallel_c,v 1.4 2006/11/08 18:25:44 bsiegert Exp $
--- backend/parallel.c.orig	Sat Jul 12 00:48:26 2008
+++ backend/parallel.c	Tue Oct 21 22:56:22 2008
@@ -569,14 +569,11 @@ list_devices(void)
 
   for (i = 0; i < 3; i ++)
   {
-    sprintf(device, "/dev/lpt%d", i);
-    if ((fd = open(device, O_WRONLY)) >= 0)
-    {
-      close(fd);
-      printf("direct parallel:%s \"Unknown\" \"Parallel Port #%d (interrupt-driven)\"\n", device, i + 1);
-    }
+    /* The lpt devices have problems under MirOS / OpenBSD, so we do not
+     * include them.
+     */
 
-    sprintf(device, "/dev/lpa%d", i);
+    snprintf(device, sizeof(device), "/dev/lpa%d", i);
     if ((fd = open(device, O_WRONLY)) >= 0)
     {
       close(fd);
